services/src/main/java/org/keycloak/userprofile/config/UPConfigUtils.java

```diff
@@ -73,24 +75,45 @@ public static UPConfig readConfig(InputStream is) throws IOException {
     }
 
     /**
-     * Validate object representation of the configuration. Validations:
+     * Validate object representation of the attribute group configuration. Validations:
      * <ul>
-     * <li>defaultProfile is defined and exists in profiles
-     * <li>parent exists for type
-     * <li>type exists for attribute
-     * <li>validator (from Validator SPI) exists for validation and it's config is correct
+     * <li>group has a name != null</li>
      * </ul>
      *
-     * @param session to be used for Validator SPI integration
      * @param config to validate
      * @return list of errors, empty if no error found
      */
-    public static List<String> validate(KeycloakSession session, UPConfig config) {
-        List<String> errors = new ArrayList<>();
+    public static List<String> validateAttributeGroups(UPConfig config) {
```

Comment:
Do we really need two separate validation methods? Looks they are called from one place, so I'd say we can have only one `validate()` as before doing all necessary validations.

Reply:
yep, makes sense. I'll create a validate() method with calls the two methods and aggregates the result.

Reply:
thanks